src/lib.rs

```diff
@@ -1461,7 +1461,11 @@ impl Connection {
         //
         // TODO: buffer packets instead of discarding as an optimization.
         if hdr.ty == packet::Type::Short && !self.is_established() {
-            return Ok(b.len());
+            return Err(drop_pkt_on_err(
+                Error::InvalidPacket,
+                self.recv_count,
+                &self.trace_id,
+            ));
```

Comment:
Now it doesn't return the length. Is there any side effect?

Reply:
So, this is s a bit tricky. I have another PR coming later with additional changes related to this, though I need to do some more tests and I split them to make it easier to review them.

The problem here is that `Ok(b.len())` and `Err(Error::Done)` should mean the same thing in theory, that is "ignore the rest of the UDP datagram".

However, over time `Err(Error::Done)` has also interpreted by applications to mean "stop reading incoming packets from the socket", which is, I realized, the wrong thing to do, as the application shouldn't stop reading from the socket ever as it might lead to a deadlock as the socket might end up never being re-armed by the event loop.

So what I want to do is:
1. Unify how packets are ignored by `recv_single()` (that is, via `drop_pkt_on_err()` and returning `Error::Done`).
2. Change `recv()` to basically treat `Error::Done` from `recv_single()` as if `Ok(b.len())` is returned (and of course update the documentation to make this clearer).

The second part means that `recv()` will not return `Error::Done` anymore so applications don't need to be changed to ignore that return value (and it also simplifies the API a bit).

(In theory we could have gone the other way, that is, alsways return `Ok(b.len())` from `recv_single()`, but that's not always easy to do).